File: internal/window.go
package internal

import (
	"os"
	"os/exec"
)

func (a *App) OpenNewInstance() error {
	executable, err := os.Executable()
	if err != nil {
		return err
	}

	cmd := exec.Command(executable)
	cmd.Env = os.Environ()

	return cmd.Start()
}

File: ui/src/fs/window.js
import {
    OpenNewInstance
} from "../../wailsjs/go/internal/App";

const spawnInstance = async () => {
    try {
        await OpenNewInstance();
    } catch (error) {
        console.error("Error spawning new instance:", error);
        throw error;
    }
};

export { spawnInstance };
